src/vm.rs

```diff
@@ -1650,6 +1650,16 @@ enum MmioEmulationError {
     InvalidAddress(u64),
 }
 
+impl core::fmt::Display for MmioEmulationError {
```

Comment:
I don't think this is sufficient; you'll need to also print the error using `{}` instead of `{:?}` to get the formatting implemented here.

Reply:
Interesting...I wasn't aware of this subtlety until now...had assumed that {:?} would call either the default implementation or the override...fixed.